src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticationScheme.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public class AuthenticationScheme
+    {
+        public AuthenticationScheme(string name, Type handlerType, IEnumerable<PathString> callbackPaths, IReadOnlyDictionary<string, object> settings)
+        {
+            Name = name ?? throw new ArgumentNullException(nameof(name));
```

Comment:
@davidfowl has rejected any use of this new syntax. @Eilon has not made an official ruling...